src/main/scala/com/gu/PriceRiseRequestBuilder.scala

```diff
@@ -0,0 +1,39 @@
+package com.gu
+
+import com.gu.FileImporter.PriceRise
+
+/**
+  * Build the body of the request that will actually apply the price rise.
+  */
+object PriceRiseRequestBuilder {
+  def apply(
+      subscription: Subscription,
+      currentGuardianWeeklySubscription: CurrentGuardianWeeklySubscription,
+      newGuardianWeeklyProductCatalogue: NewGuardianWeeklyProductCatalogue,
+      priceRise: PriceRise
+  ): PriceRiseRequest = {
+
+    require(currentGuardianWeeklySubscription.subscriptionNumber == priceRise.subscriptionName, "Price rise applied to a wrong subscription.")
+
+    val newGuardianWeeklyProduct =
+      GuardianWeeklyProduct(currentGuardianWeeklySubscription, newGuardianWeeklyProductCatalogue)
+
+    val chargeOverride = ChargeOverride(
+      newGuardianWeeklyProduct.productRatePlanChargeId,
+      priceRise.newPrice
+    )
+
+    val addProductRatePlan = AddProductRatePlan(
+      newGuardianWeeklyProduct.productRatePlanId,
+      priceRise.priceRiseDate,
+      Some(List(chargeOverride))
```

Comment:
So do we always provide a charge override when adding new plans? Does this work even if we are moving the user onto standard catalog pricing?

Reply:
Yes and yes. If we want to identify all the actual overrides we could query the data lake.